tests/nlu/classifiers/test_embedding_intent_classifier.py

```diff
@@ -104,3 +108,47 @@ def test_check_labels_features_exist(messages, expected):
         EmbeddingIntentClassifier._check_labels_features_exist(messages, attribute)
         == expected
     )
+
+
+def as_pipeline(*components):
+    return [{"name": c} for c in components]
+
+
+@pytest.mark.parametrize(
+    "classifier_params, output_length, output_should_sum_to_1",
+    [
+        ({}, 10, True),  # default config
+        ({"ranking_length": 0}, 10, False),  # no normalization
+        ({"ranking_length": 3}, 3, True),  # lower than default ranking_length
+        ({"ranking_length": 12}, 10, False),  # higher than default ranking_length
+    ],
+)
+async def test_softmax_normalization(
```

Comment:
let's add a test that non softmax scores doesn't get normalized

Reply:
Was going to but couldn't figure out a declarative test for it -- what can I specifically say about the (truncated) output of the margin one that tells me it wasn't normalized? Is it always the case that the sum of top 10 intents will not equal 1 and that's enough to test?

Reply:
not sure, but at least make sure that the output is of `ranking_length`